x-pack/plugins/security_solution/common/endpoint/schema/trusted_apps.ts

```diff
@@ -87,39 +87,37 @@ const MacEntrySchema = schema.object({
   ...CommonEntrySchema,
 });
 
+const entriesSchemaOptions = {
+  minSize: 1,
+  validate(entries: ConditionEntry[]) {
+    return (
+      getDuplicateFields(entries)
+        .map((field) => `duplicatedEntry.${field}`)
+        .join(', ') || undefined
+    );
+  },
+};
+
 /*
- * Entry Schema depending on Os type using schema.conditional.
+ * Entities array schema depending on Os type using schema.conditional.
  * If OS === WINDOWS then use Windows schema,
  * else if OS === LINUX then use Linux schema,
  * else use Mac schema
+ *
+ * The validate function checks there is no duplicated entry inside the array
  */
-const EntrySchemaDependingOnOS = schema.conditional(
+const EntriesSchema = schema.conditional(
   schema.siblingRef('os'),
   OperatingSystem.WINDOWS,
-  WindowsEntrySchema,
+  schema.arrayOf(WindowsEntrySchema, entriesSchemaOptions),
   schema.conditional(
     schema.siblingRef('os'),
```

Comment:
@elastic/security-solution `joi`'s reference implementation is now less permissive.

You were using a reference from the schema of the type of an array to reference a sibling of the whole array's type

```ts
{
   os: schema.something(),
   entries: schema.arrayOf(schema.conditional(schema.siblingRef('os'), [...]))
}
```

This is no longer allowed, so I had to move the conditional to wrap the whole `arrayOf` type instead.